corehq/apps/export/utils.py

```diff
@@ -596,21 +597,22 @@ def migrate_domain(domain, dryrun=False, force_convert_columns=False):
     metas = []
     if export_count:
         for old_export in with_progress_bar(
-                stale_get_exports_json(domain),
+                get_exports_json(domain),
                 length=export_count,
                 prefix=domain):
-            try:
-                _, migration_meta = convert_saved_export_to_export_instance(
-                    domain,
-                    SavedExportSchema.wrap(old_export),
-                    dryrun=dryrun,
-                    force_convert_columns=force_convert_columns,
-                )
-            except Exception as e:
-                print('Failed parsing {}: {}'.format(old_export['_id'], e))
-                raise e
-            else:
-                metas.append(migration_meta)
+            with CriticalSection(['saved-export-{}'.format(old_export._id)], timeout=120):
+                try:
+                    _, migration_meta = convert_saved_export_to_export_instance(
+                        domain,
+                        SavedExportSchema.wrap(old_export),
```

Comment:
does this `old_export` get saved throughout this process? if so i think you still might have an issue since the entire get-and-save operation needs to be under the `CriticalSection`.  as it is now, `get_exports_json` is outside the `CriticalSection` so between the time that is called and this executes, something else could have saved the object. One way to fix would be to call `.get()` to get the latest object again under the `CriticalSection`